tools/rsdlEditor/README.md

```diff
@@ -0,0 +1,84 @@
+# RSDL Editor
+
+An editor component that provides syntax highlighting for writing RSDL schemas.
+
+## Installation
+
+Then run:
+
+```
+npm install
+```
+
+Then run:
+
+```
+npm run build
+```
+
+The build command will store the output files in the `dist` folder.
+
+**Note**: You need to run the build command before setting up packages that depend on this library. This package is not on npm, so the dependent
+packages will fetch this package from the local filesystem as is.
+
+To add this package as a depedency to another project, add reference to this library to
+the other project's `package.json` `dependencies`, the reference should be a file
+path that points to this package's `dist` folder:
+
+```json
+"dependencies": {
+    "rsdl-editor": "file:path/to/rsdlEditor
+}
+```
+
+## Usage
+
+```ts
+import { initRsdlEditor } from "rsdl-editor";
+
+const editor = initRsdlEditor(document.getElementById("editor"), onRsdlUpdated);
+// schema is a string in rsdl format
+const schema: string = "...";
+editor.updateContent(schema);
+
+// this handler will be called when the user updates the content in the editor
+function onUrlUpdated(content: string) {
+    console.log("URL has been updated to", content);
+}
+
+// gets the current contents of the editor
+const currentContent = editor.getContent();
+
+```
+
+You can also load this library directly in a html's `src` tag. See the [example.html](./example.html) file for a sample.
+
+## Notes for future reference (feel free to ignore this)
```

Comment:
When I try to run the "npm run build" command I get the error "Cannot use import outside a module" in rollup.config.js.

Reply:
Which version of nodejs are you using?

Reply:
[Rollup dcoumentation](https://rollupjs.org/guide/en/#configuration-files) says that it transpiles the config file before running it, not sure why it doesn't work on your end. I've updated the file to be a CommonJs file that uses `require` instead of `import`. Let me know if this works.

Reply:
Wanted to add that we found that on Windows using Powershell instead of the CMD prompt may yield better results. This was at least the case for @mikepizzo. On my machine things were more consistent between cmd and Powershell.

Reply:
That explains why I didn't have this issue, I'm using PowerShell 7.2.1